File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        cost=0
        curMax=0
        for i in range(len(colors)):
            if(i>0 and colors[i]!=colors[i-1]):
                curMax=0
            cost+=min(curMax,neededTime[i])
            curMax=max(curMax,neededTime[i])
        return cost

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
     
//         int cost=0;
//         for(int i=1;i<neededTime.size();++i)
//         {
//             int temp=0;
//             if(colors[i]==colors[i-1])
//                 temp=min(neededTime[i],neededTime[i-1]);
//             cost+=temp;
//         }
//         return cost;
        
        int totalTime = 0, currMaxTime = 0;
        for (int i = 0; i < colors.size(); ++i) {
            if (i > 0 && colors[i] != colors[i - 1]) {
                currMaxTime = 0;
            }
            totalTime += min(currMaxTime, neededTime[i]);
            
            currMaxTime = max(currMaxTime, neededTime[i]);
           // cout<<"i : "<<i<<" totalTime :  "<<totalTime<<" currMaxTime : "<<currMaxTime<<endl;
        }
        return totalTime;
    }
};